File: MATTHEW/leetcode/AIcoding/5.Qwen-3/python3_solutions/2239-find-closest-number-to-zero.py
class Solution:
    def closestToZero(self, nums: List[int]) -> int:
        closest = nums[0]
        for num in nums[1:]:
            if abs(num) < abs(closest):
                closest = num
            elif abs(num) == abs(closest):
                if num > closest:
                    closest = num
        return closest

File: MATTHEW/leetcode/AIcoding/5.Qwen-3/rust_solutions/2239-find-closest-number-to-zero.rs
struct Solution;

impl Solution {
    pub fn closest_to_zero(nums: Vec<i32>) -> i32 {
        let mut closest = nums[0];
        for &num in &nums {
            if num.abs() < closest.abs() || (num.abs() == closest.abs() && num > closest) {
                closest = num;
            }
        }
        closest
    }
}